spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -67,4 +73,32 @@ public Subscriber getSubscriber(String subscriptionName, MessageReceiver receive
 				.setCredentialsProvider(this.credentialsProvider)
 				.build();
 	}
+
+	@Override
+	public PullRequest getPullRequest(String subscriptionName, Integer maxMessages,
+			Boolean returnImmediately) {
+		PullRequest.Builder pullRequestBuilder =
+				PullRequest.newBuilder().setSubscriptionWithSubscriptionName(
+						SubscriptionName.of(this.projectId, subscriptionName));
+
+		if (maxMessages != null) {
+			pullRequestBuilder.setMaxMessages(maxMessages);
+		}
+
+		if (returnImmediately != null) {
+			pullRequestBuilder.setReturnImmediately(returnImmediately);
+		}
+
+		return pullRequestBuilder.build();
+	}
+
+	@Override
+	public SubscriptionAdminSettings getSubscriptionAdminSettings() {
+		try {
+			return SubscriptionAdminSettings.newBuilder().build();
```

Comment:
Looks like this can go to the interface as a `default` method.

Reply:
Not really, because I'll be adding some timeout logic here.